File: garnet/bin/memory_monitor/test/test_utils.cc
// Copyright 2019 The Fuchsia Authors. All rights reserved.
// Use of this source code is governed by a BSD-style license that can be
// found in the LICENSE file.

#include "garnet/bin/memory_monitor/test/test_utils.h"

namespace memory {

// static.
void TestUtils::CreateCapture(memory::Capture& capture,
                              const CaptureTemplate& t) {
  capture.time_ = t.time;
  capture.kmem_= t.kmem;
  for (auto vmo : t.vmos) {
    capture.koid_to_vmo_.emplace(vmo.koid, vmo);
  }
  for (auto process : t.processes) {
    capture.koid_to_process_.emplace(process.koid, process);
  }
}

// static.
std::vector<ProcessSummary> TestUtils::GetProcessSummaries(
    const Summary& summary) {
  std::vector<ProcessSummary> summaries = summary.process_summaries();
  sort(summaries.begin(), summaries.end(),
       [](ProcessSummary a, ProcessSummary b) {
    return a.koid() < b.koid();
      });
  return summaries;
}
}  // namespace memory

File: src/connectivity/wlan/lib/mlme/rust/src/client/test_utils.rs
// Copyright 2021 The Fuchsia Authors. All rights reserved.
// Use of this source code is governed by a BSD-style license that can be
// found in the LICENSE file.

#![cfg(test)]

use {
    crate::client::{EventId, TimedEvent, TimedEventClass},
    std::collections::HashMap,
    wlan_common::timer::TimeStream,
};

pub fn drain_timeouts(
    time_stream: &mut TimeStream<TimedEvent>,
) -> HashMap<TimedEventClass, Vec<(TimedEvent, EventId)>> {
    let mut timeouts = HashMap::new();
    loop {
        match time_stream.try_next() {
            Ok(Some((_, timed_event))) => {
                timeouts
                    .entry(timed_event.event.class())
                    .or_insert(vec![])
                    .push((timed_event.event, timed_event.id));
            }
            _ => return timeouts,
        };
    }
}
